diff --git a/node_modules/keycloak-connect/middleware/auth-utils/grant-manager.js b/node_modules/keycloak-connect/middleware/auth-utils/grant-manager.js
index 09054d3..c247bf3 100644
--- a/node_modules/keycloak-connect/middleware/auth-utils/grant-manager.js
+++ b/node_modules/keycloak-connect/middleware/auth-utils/grant-manager.js
@@ -424,8 +424,8 @@ GrantManager.prototype.validateToken = function validateToken (token, expectedTy
       reject(new Error('invalid token (wrong type)'))
     } else if (token.content.iat < this.notBefore) {
       reject(new Error('invalid token (stale token)'))
-    } else if (token.content.iss !== this.realmUrl) {
-      reject(new Error('invalid token (wrong ISS)'))
+    } else if (process.env.KEYCLOAK_ISSUER_IGNORE !== 'true' && token.content.iss !== this.realmUrl) {
+      reject(new Error('invalid token (wrong ISS) Expected ' + this.realmUrl + ' but got ' + token.content.iss + ' instead'))
     } else {
       const audienceData = Array.isArray(token.content.aud) ? token.content.aud : [token.content.aud]
       if (expectedType === 'ID') {
